src/api/app/models/bs_request.rb

```diff
@@ -154,6 +154,9 @@ def self.new_from_xml(xml)
     if hashed['id']
       theid = hashed.delete('id') { raise 'not found' }
       theid = Integer(theid)
+      if BsRequest.find_by_number(theid)
```

Comment:
modifing user data in validation is IMHO not a good thing (in the commentor case below).

So, I keep this code but add validations in addition to be save, okay?